src/collection-facets/facets-more-content.ts

```diff
@@ -0,0 +1,310 @@
+/* eslint-disable dot-notation */
+import { css, CSSResultGroup, html, LitElement, PropertyValues } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { Bucket, SearchParams } from '@internetarchive/search-service';
+
+import { CollectionNameCacheInterface } from '@internetarchive/collection-name-cache';
+import { SelectedFacets, FacetGroup, defaultSelectedFacets } from '../models';
+import { LanguageCodeHandlerInterface } from '../language-code-handler/language-code-handler';
+
+@customElement('facets-more-content')
+export class FacetsMoreContent extends LitElement {
+  @property({ type: String }) facetKey?: string;
+
+  @property({ type: String }) facetAggregationKey?: string;
+
+  @property({ type: String }) fullQuery?: string;
+
+  @property({ type: Object }) modalManager?: any;
+
+  @property({ type: Object }) searchService?: any;
+
+  @property({ type: Object }) selectedFacets?: SelectedFacets;
+
+  @property({ type: Object, attribute: false }) aggr = [];
+
+  @property({ type: Object }) castedBuckets?: Bucket[] = [];
+
+  @property({ type: Object }) selectedFacet?: String;
+
+  @property({ type: Object })
+  languageCodeHandler?: LanguageCodeHandlerInterface;
+
+  @property({ type: Object }) allFacetGroups?: FacetGroup[] = [];
+
+  @state() pageNumber = 1;
+
+  @state() loading = 1;
+
+  @property({ type: Object })
+  collectionNameCache?: CollectionNameCacheInterface;
+
+  private facetsPerPage = 25;
+
+  async updated(changed: PropertyValues) {
+    if (changed.has('facetKey')) {
+      this.loading = 1;
+      this.pageNumber = 1;
+
+      await this.fetchSpecificFacets(
+        this.facetAggregationKey as unknown as string
+      );
+      this.loading = 0;
+    }
+
+    if (changed.has('aggr')) {
+      this.filterFacets();
+    }
+  }
+
+  async fetchSpecificFacets(facetAggregationKey: string) {
+    const aggregations = {
+      advancedParams: [
+        {
+          field: facetAggregationKey,
+          size: 1000, // todo - ?
+        },
+      ],
+    };
+
+    const params: SearchParams = {
+      query: this.fullQuery as string,
+      fields: ['identifier'],
+      aggregations,
+      rows: 1, // todo - ?
+    };
+
+    const results = await this.searchService?.search(params);
+    this.aggr = results?.success?.response.aggregations;
+  }
+
+  async filterFacets() {
+    Object.entries(this.aggr ?? []).forEach(([key, buckets]) => {
+      if (key === 'year_histogram') return;
+
+      this.castedBuckets = buckets['buckets'] as Bucket[];
+    });
+  }
+
+  private get renderPaginations() {
+    const paging = [];
+
+    const lenght = Object.keys(this.castedBuckets as []).length;
+    const numberOfPages = Math.ceil(lenght / this.facetsPerPage);
+
+    const onClickEvent = (e: Event) => {
+      this.pageClick(e);
+    };
+
+    let page = 1;
+    for (page = 1; page <= numberOfPages; page += 1) {
+      if (this.pageNumber === page) {
```

Comment:
looks like u only need 1 extra class, how about something like: 
```suggestion
      const currentPageClass = this.pageNumber === page ? 'current-page' : '';
      const button = html`button
             class=${`page-number ${currentPageClass}`}
             @click="${onClickEvent}"
           >
             ${page}
           </button>`
```

Reply:
Changed it with differently